docs/sdk2/usage/CART_DISCOUNT_SYNC.md

```diff
@@ -0,0 +1,390 @@
+
+##### Syncing from a commercetools project
+
+When syncing from a source commercetools project, you can use [`toCartDiscountDrafts`](#todo)
+method that transforms(resolves by querying and caching key-id pairs) and maps from a `CartDiscount` to `CartDiscountDraft` using cache in order to make them ready for reference resolution by the sync, for example: 
+
+````java
+// Build a ByProjectKeyCartDiscountsGet for fetching cart discounts from a source CTP project without any references expanded for the sync:
+final CartDiscountQuery cartDiscountQuery = client.categories().get();
+
+// Query all cart discounts (NOTE this is just for example, please adjust your logic)
+final List<CartDiscount> cartDiscounts = QueryUtils.queryAll(byProjectKeyCartDiscountsGet,
+        (cartDiscounts) -> cartDiscounts)
+        .thenApply(lists -> lists.stream().flatMap(List::stream).collect(Collectors.toList()))
+        .toCompletableFuture()
+        .join();
+````
+
+In order to transform and map the `CartDiscount` to `CartDiscountDraft`, 
+Utils method `toCartDiscountDrafts` requires `sphereClient`, implementation of [`ReferenceIdToKeyCache`](#todo) and `cartDiscounts` as parameters.
+For cache implementation, You can use your own cache implementation or use the class in the library - which implements the cache using caffeine library with an LRU (Least Recently Used) based cache eviction strategy[`CaffeineReferenceIdToKeyCacheImpl`](#todo).
+Example as shown below:
+
+````java
+//Implement the cache using library class.
+final ReferenceIdToKeyCache referenceIdToKeyCache = new CaffeineReferenceIdToKeyCacheImpl();
+
+//For every reference fetch its key using id, cache it and map from CartDiscount to CartDiscountDraft. With help of the cache same reference keys can be reused.
+CompletableFuture<List<CartDiscountDraft>> cartDiscountDrafts = CartDiscountTransformUtils.toCartDiscountDrafts(client, referenceIdToKeyCache, cartDiscounts);
+````
+
+##### Syncing from an external resource
+
+- When syncing from an external resource, `ResourceIdentifier`s with their `key`s have to be supplied as following example:
+
+```` java
+final CartDiscountDraft cartDiscountDraft = CartDiscountDraftBuilder.of()
+            .cartPredicate("cartPredicate")
+            .name(LocalizedString.ofEnglish("foo"))
+            .requiresDiscountCode(true)
+            .sortOrder("0.1")
+            .target(CartDiscountTargetBuilder::shippingBuilder)
+            .value(CartDiscountValueAbsoluteDraftBuilder.of()
+                    .money(
+                            MoneyBuilder.of()
+                                    .centAmount(10L)
+                                    .currencyCode(DefaultCurrencyUnits.EUR.getCurrencyCode())
+                                    .build())
+                    .build())
+            .key("cart-discount-key")
+            .custom(CustomFieldsDraft.ofTypeKeyAndJson("type-key", emptyMap())) // note that custom type provided with key
```

Comment:
The method `CustomFieldsDraft.ofTypeKeyAndJson` isn't valid vor SDK v2. Its CustomFieldsDraftBuilder.of()

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/1030/commits/d458876d5189a38b3c0c0ad325b2de1292712ad4